### ISSUE TITLE
[Request] Publish to npm/bower registries

### ISSUE THREAD

**Issue by chimericdream** [jgraph/mxgraph#24]
I have a Node.js application that I am working on which will have some interactive diagramming. I have used draw.io in the past, so I like the idea of embedding mxgraph. However, I'd rather it be a standard node or bower dependency rather than something I manually download and commit to the repo.

**Comment by alderg** [comment]
Thanks for your input, we'll pass this on to the product team.

**Comment by chimericdream** [comment]
I know this is a delayed response, but why was the issue closed? I'm glad to hear that the product team might be looking into this, but I would have expected the issue to remain open until either the module is pushed to bower/npm or you decide not to publish it. Closing the issue before either of those happens implies that it is resolved. Leaving it open makes it easier for users (in this case just me, but hey, it _could_ be other people... 😁 ) to be notified when this does in fact get completed (or closed as `wontfix`). Otherwise, it becomes the responsibility of a user to come here to check if bugs, issues, or other requests are resolved.

**Comment by chimericdream** [comment]
I'll also note that I would be glad to fork the repo and add the `package.json` and `bower.json` files myself, then submit a PR. I completely understand that this might be a lower priority for your team compared to how much I want the feature. But having only published a couple modules, I don't know what the process is if I run the actual `npm publish` (or `bower publish`) command but want the jgraph team to retain ownership. If it's easy to transfer ownership to you once the modules are published, I don't mind doing the work (did I mention I _really_ want to use this?). That's the whole point of open source, after all.

**Comment by davidjgraph** [comment]
Github isn't our primary issue tracker at this point, so we're not able to use in a conventional way. We're still discussing whether to leave Github issues open, since it might cause confusion.

In terms of a PR, that's definitely our preferred route, rather than holding an issue open. We've never published a module, so you're winning in terms of experience :).

**Comment by chimericdream** [comment]
@davidjgraph I did a little digging, and it looks like publishing to NPM is something I can do, since adding/removing owners is pretty simple. I won't publish it to Bower, though. Transferring ownership looks like it is currently rather difficult.

Once everything is in place, though, you should just be able to run `bower register mxgraph git://github.com/jgraph/mxgraph.git`.

**Comment by leungwensen** [comment]
+1

besides, is there any plan to rewrite the project with ES2016+?  If there is, https://github.com/lebab/lebab might help

**Comment by MojoJojo** [comment]
+1 for ES2016

**Comment by davidjgraph** [comment]
Please keep to one subject per issue. This issue is now a mess of various issues.

**Comment by davidjgraph** [comment]
It looks like someone has done the NPM publish for the JS part of mxGraph, https://www.npmjs.com/package/mxgraph-js.

If anyone wants bower, please open as another issue.

**Comment by davidjgraph** [comment]
https://www.npmjs.com/package/mxgraph, does that satisfy the issue?

**Comment by chimericdream** [comment]
Yep, that'll work.

**Comment by davidjgraph** [comment]
This pull request, https://github.com/jgraph/mxgraph/pull/63, has example usage if needed.
